### PR TITLE
limit bandwidth + assert records in test

### LINKED ISSUE #2: use bwlimit
### Feature description

use bwlimit to limit connection bandwidth https://github.com/conduitio/bwlimit

### PR DESCRIPTION
**Pull request by maha-hajja**
### Description

Fixes #2 

### Quick checks:

- [x] There is no other [pull request](https://github.com/conduitio-labs/conduit-connector-grpc-client/pulls) for the same update/change.
- [ ] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: destination.go [hunk 1]
```diff
@@ -43,6 +45,8 @@ type Destination struct {
 type Config struct {
 	// url to gRPC server
 	URL string `json:"url" validate:"required"`
+	// the bandwidth limit in bytes/second
+	RateLimit int `json:"rateLimit" default:"10000"`
```
Comment (hariso) [inline destination.go:49]: How is the rate limit disabled? By setting it to a high number or ?

File: destination_test.go [hunk 2]
```diff
@@ -104,18 +106,28 @@ func startTestServer(t *testing.T, lis net.Listener) {
...
 					// read from the stream to simulate receiving data from the client
-					req, err := stream.Recv()
+					rec, err := stream.Recv()
 					if err == io.EOF {
 						return nil
 					}
 					if err != nil {
 						return err
 					}
+					// convert the sdkRec to sdk.Record to compare with expected records
```
Comment (hariso) [inline destination_test.go:119]: Nitpick: convert the gRPC record to sdk.Record?

File: destination.go (hunk 1 above)
Comment (maha-hajja) [reply destination.go:49]: I created this before we talked about disabling it 👀 this was the default value, will check how to disable if possible

File: destination.go [hunk 3]
```diff
@@ -68,12 +72,25 @@ func (d *Destination) Configure(ctx context.Context, cfg map[string]string) erro
...
+			bwgrpc.WithBandwidthLimitedContextDialer(bwlimit.Byte(d.config.RateLimit), bwlimit.Byte(d.config.RateLimit), d.dialer),
+		)
+	} else {
+		conn, err = grpc.DialContext(ctx,
+			d.config.URL,
+			grpc.WithContextDialer(d.dialer),
+			grpc.WithInsecure(), //nolint:staticcheck // todo: will use mTLS with connection
+			grpc.WithBlock(),
+		)
+	}
```
Comment (hariso) [inline destination.go:93]: Nitpick ahead. I'm thinking about this code in the following: specific options are always there (WithInsecure, WithContextDialer...). If rate limiting is enabled, we add it on top of that. If mTLS is enabled, we add it on top of that.

If mTLS is enabled, we need to add to both blocks for example. What might be a good option as well, it to have a slice of options, and do something like:
```
options := WithContextDialer, WithBlock // standard ones
if rateLimitingEnabled {
    options = append(options, WithRateLimiting)
}
if mTLS {
    options = append(options, WithMTLS)  
}
```

**Review by hariso — APPROVED** [review]

File: paramgen_dest.go [hunk 4]
```diff
@@ -9,6 +9,14 @@ import (
...
 func (Config) Parameters() map[string]sdk.Parameter {
 	return map[string]sdk.Parameter{
+		"rateLimit": {
+			Default:     "0",
+			Description: "the bandwidth limit in bytes/second, use \"0\" to disable rate limiting.",
+			Type:        sdk.ParameterTypeInt,
+			Validations: []sdk.Validation{
+				sdk.ValidationGreaterThan{Value: -1},
+			},
+		},
```
[the remark this answers is not in the document]
Comment (maha-hajja) [reply paramgen_dest.go:19]: yess feels like magic 🪄

File: destination.go (hunk 3 above)
Comment (maha-hajja) [reply destination.go:93]: oh I love this! I had the same idea at the beginning but tried to append it to the connection after it was created, which didn't work, but your way should! will try it out
thanks for the suggestions!

Comment (hariso) [reply destination.go]: You're welcome!

**Pull request merged**